debezium-core/src/main/java/io/debezium/heartbeat/DatabaseHeartbeatImpl.java

```diff
@@ -34,21 +35,27 @@
 
     private final String heartBeatActionQuery;
     private final JdbcConnection jdbcConnection;
+    private final HeartbeatErrorHandler errorHandler;
 
-    DatabaseHeartbeatImpl(Configuration configuration, String topicName, String key, JdbcConnection jdbcConnection, String heartBeatActionQuery) {
-        super(configuration, topicName, key);
+    DatabaseHeartbeatImpl(Duration heartbeatInterval, String topicName, String key, JdbcConnection jdbcConnection, String heartBeatActionQuery,
```

Comment:
I decoupled Heartbeat from the dependency to Configuration. Instead `heartbeatInterval` is passed explicitly (and `heartbeatQuery` where applicable).